pwa/components/admin/Admin.tsx

```diff
@@ -116,6 +117,7 @@ const AdminWithContext = ({ session }: { session: Session }) => {
 const AdminWithOIDC = () => {
   // Can't use next-auth/middleware because of https://github.com/nextauthjs/next-auth/discussions/7488
   const { data: session, status } = useSession();
+  const { permissions } = usePermissions();
```

Comment:
usePermissions can only be called by children of react-admin's `<Admin>` component (otherwise there is no AuthProviderContext and QueryClientContext available). That's why you have an error. 

The proper fix would be to call the `usePermissions` in `BookList` and other admin-only components, and redirect to a `/forbidden` CustomRoute if the permissions don't match.